historydag/dag.py

```diff
@@ -397,23 +397,32 @@ def copy(self) -> "HistoryDag":
         """
         return pickle.loads(pickle.dumps(self))
 
-    def merge(self, other: "HistoryDag"):
-        r"""Graph union this history DAG with another."""
+    def merge(self, trees: List["HistoryDag"]):
+        r"""Graph union this histroy DAG with all those in a list of history DAGs."""
         selforder = self.postorder()
-        otherorder = other.postorder()
-        # hash and __eq__ are implemented for nodes, but we need to retrieve
-        # the actual instance that's the same as a proposed node-to-add:
         nodedict = {n: n for n in selforder}
-        for n in otherorder:
-            if n in nodedict:
-                pnode = nodedict[n]
-            else:
-                pnode = n.node_self()
-                nodedict[n] = pnode
 
-            for _, edgeset in n.clades.items():
-                for child, weight, _ in edgeset:
-                    pnode.add_edge(nodedict[child], weight=weight)
+        if not isinstance(trees, list):
```

Comment:
One more suggestion. It would be nice if merge could accept any iterable containing `HistoryDag` objects to merge. For instance, we might have them (for some reason) in a tuple, or more likely, a lazily evaluated generator object.
To make this happen, it would be better to test if `isinstance(trees, HistoryDag)`, and then otherwise assume that `trees` is some iterable container containing `HistoryDag`s.

Reply:
It won't matter to the type checker, but to express that we don't _need_ the iterable container to be a list, the input type should actually be `Union["HistoryDag", Sequence["HistoryDag"]]`.

Reply:
Oh yeah that's a good point! Making changes now